### PR TITLE
Add sterr to output list

### PR DESCRIPTION
**Pull request by noamblitz**
### Changes

Also save sterr to bytes for docker boefjes

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3559

### Demo

_Please add some proof in the form of screenshots or screen recordings to show (off) new functionality, if there are interesting new features for end-users._

### QA notes

_Please add some information for QA on how to test the newly created code._

---

### Code Checklist

- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by noamblitz** [comment]
https://github.com/minvws/nl-kat-coordination/issues/3559#issuecomment-2400023800

Important to think about.

File: boefjes/boefjes/plugins/kat_nmap_tcp/main.py [hunk 1]
```diff
@@ -18,4 +18,11 @@ def run(boefje_meta: dict):
         cmd.append("-6")
 
     cmd.extend(["-oX", "-", str(ip)])
-    return [({"openkat/nmap-output"}, subprocess.run(cmd, capture_output=True).stdout.decode())]
+
+    output = subprocess.run(cmd, capture_output=True)
+    results = [({"openkat/nmap-output"}, output.stdout.decode())]
+
+    if output.stderr:
+        return results.append(({"error/boefje"}, output.stderr.decode()))
```
[the remark this answers is not in the document]
Comment (noamblitz) [reply boefjes/boefjes/plugins/kat_nmap_tcp/main.py:26]: Hmm yeah I think you are right. Lets do that.

Comment (underdarknl) [inline boefjes/boefjes/plugins/kat_nmap_tcp/main.py]:
```suggestion
        if not output.stdout:
            results = []
        return results.append(({"error/boefje"}, output.stderr.decode()))
```

Comment (noamblitz) [reply boefjes/boefjes/plugins/kat_nmap_tcp/main.py]: I also made a push. That should also cover this, right?

Comment (underdarknl) [reply boefjes/boefjes/plugins/kat_nmap_tcp/main.py]: Your solution makes sure we don't Delete any previous proofs. 
My suggestions allows for a situation where we still have some logical output that we'd want to ingest into the graph, but also some errors (which did not stop us from creating data). The million dollar question is: 'is there a situation where nmap throws errors, but still manages to produce valid output?'

Comment (noamblitz) [reply boefjes/boefjes/plugins/kat_nmap_tcp/main.py]: Haha yeah good question. To be honest I don't think so. Im more inclined to be on the safe side and never delete old observations. If you feel like there are situations when nmap throws errors & good output, then I'm happy to rewrite it to your solution.

**Review by ammar92 — APPROVED** [review]
No remarks

**Comment by dekkers** [comment]
I tested it and verified that the boefje still works. If I introduce an error the boefje will get failed status.

The stderr of the command is not saved as is mentioned in the description because the exception is not catched. But this part is already useful, so I am merging the PR and creating a new issue.

**Review by dekkers — APPROVED** [review]

**Pull request merged**